test/e2e/egress.go

```diff
@@ -81,37 +82,37 @@ var _ = framework.KubeDescribe("Static Egress creation", func() {
 		defer func() {
 			By("deleting the configmap")
 			defer GinkgoRecover()
-			err2 := cs.Core().ConfigMaps(ns).Delete(cmap.Name, metav1.NewDeleteOptions(0))
+			err2 := cs.CoreV1().ConfigMaps(ns).Delete(cmap.Name, metav1.NewDeleteOptions(0))
 			Expect(err2).NotTo(HaveOccurred())
 		}()
-		_, err = cs.Core().ConfigMaps(ns).Create(cmap)
+		_, err = cs.CoreV1().ConfigMaps(ns).Create(cmap)
 		Expect(err).NotTo(HaveOccurred())
 
 		// wait for egress route and NAT GWs ready and POD exit code 0 vs 2
-		for {
-			p, err := cs.Core().Pods(ns).Get(pingPod.Name, metav1.GetOptions{})
-			if err != nil {
-				Expect(fmt.Errorf("Could not get POD %s", pingPod.Name)).NotTo(HaveOccurred())
-				return
-			}
-
-			if p.Status.ContainerStatuses[0].State.Terminated == nil {
-				time.Sleep(10 * time.Second)
-				continue
-			}
-
-			switch n := p.Status.ContainerStatuses[0].State.Terminated.ExitCode; n {
-			case 0:
-				return
-			case 2:
-				// set error
-				Expect(fmt.Errorf("failed to change public IP")).NotTo(HaveOccurred())
-				return
-			}
-		}
+		Eventually(func() error {
```

Comment:
Does `Eventually` retry?

I think the problem was that it takes a while to actually have everything setup before you get the routing change applied.

Reply:
Yes, it'll retry every 10 seconds and time out after 10 minutes (see https://github.com/zalando-incubator/kubernetes-on-aws/pull/2385/files#diff-b9184f93ce39a9ec0e9abba3438a9650R94).